src/poly/dcrt/matrix/mmap_matrix.rs

```diff
@@ -66,12 +66,22 @@ impl<T: MmapMatrixElem> MmapMatrix<T> {
 
     pub fn block_entries(&self, rows: Range<usize>, cols: Range<usize>) -> Vec<Vec<T>> {
         let entry_size = self.entry_size();
+        let page_size = unsafe { libc::sysconf(libc::_SC_PAGESIZE) as usize };
         parallel_iter!(rows)
             .map(|i| {
-                let offset = entry_size * (i * self.ncol + cols.start);
-                let mmap = map_file(&self.file, offset, entry_size * cols.len());
-                let row_vec = mmap.to_vec();
-                let row_col_vec = row_vec
+                let raw_offset = entry_size * (i * self.ncol + cols.start);
+                let aligned_offset = raw_offset - (raw_offset % page_size);
+                let offset_adjustment = raw_offset - aligned_offset;
+                let required_size = offset_adjustment + entry_size * cols.len();
+                let mapping_size = if required_size % page_size == 0 {
+                    required_size
+                } else {
+                    ((required_size / page_size) + 1) * page_size
+                };
+                let mmap = map_file(&self.file, aligned_offset, mapping_size);
+                let row_data =
+                    &mmap[offset_adjustment..offset_adjustment + entry_size * cols.len()];
```

Comment:
I think this might be too large if the column size is large.

Reply:
> original length of byte is entry_size * cols.len() 

Ah yes, this is true.
But in this case I think the original implementation on main branch needs to be fixed.

Reply:
Sure i can investigate on seperate pr, its also might related to during middle param test, while sampling k_preimage we concat column for 1330 columns and this operation takes 1.5hr